src/bcccp/carpark/paystation/PaystationController.java

```diff
@@ -20,14 +20,12 @@
 	
 
 	public PaystationController(ICarpark carpark, IPaystationUI ui) {
-		//TODO Implement constructor
-		try {
-		this.carpark = carpark;
-		this.ui = ui;
-		this.state = State.IDLE;
-		}catch(NullPointerException e) {
-			display("Can't be null");
-		}
+		
+		this.carpark_ = carpark;
+		this.ui_ = ui;
+		
+		ui.registerController(this);		
```

Comment:
Where is the exception?

Reply:
probably you can throw an exception here:
```
if (carpark = null ) {
throw new NullPointerException("Carpark ID can't be null);
}
```